frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -64,9 +72,112 @@ import { CustomResourceDefinitionModel } from '@console/internal/models';
+  />
+);
+
+type WrappedToggleViewProps = {
+  toggleValue: ToggleValue;
+  toggleView?: boolean;
+  title?: string;
+  onChangeToggleValue: (newValue: ToggleValue) => void;
+};
+const WrappedToggleView: React.FC<WrappedToggleViewProps> = ({
+  toggleValue,
+  // toggleView,
+  title,
+  onChangeToggleValue,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <div className="co-operator-details__toggle-view">
+        <span className="co-operator-details__toggle-view-heading">
+          <SectionHeading text={title || t('olm~All Instances')} />
+        </span>
+        <ToggleView
+          value={toggleValue}
+          onChange={onChangeToggleValue}
+          label={t('olm~Show operands in:')}
+          optionTitles={[t('olm~All namespaces'), t('olm~Current namespace only')]}
+        />
+      </div>
+      {/* 
+      {toggleView && (
+        <div className="co-operator-details__alert">
+          <Alert
+            isInline
+            className="co-alert co-break-word"
+            variant="info"
+            title={t('olm~Operands from all managed namespaces are being displayed.')}
+          />
+        </div>
+      )} */}
+    </>
+  );
+};
+
+const firehoseResources = (models, providedAPIs: ProvidedAPI[], namespaced: boolean) => {
+  // Exclude provided APIs that do not have a model
+  return providedAPIs.reduce((resourceAccumulator, api) => {
+    const reference = referenceForProvidedAPI(api);
+    const model = models?.[reference];
+    return model
+      ? [
+          ...resourceAccumulator,
+          namespaced
+            ? {
+                kind: referenceForProvidedAPI(api),
+                namespaced: model.namespaced,
+                prop: api.kind,
+              }
+            : {
+                kind: referenceForProvidedAPI(api),
+                prop: api.kind,
+              },
+        ]
+      : resourceAccumulator;
+  }, []);
+};
+
+const flattenResources = (providedAPIs: K8sResourceKind[]) => {
+  const owners = (ownerRefs: OwnerReference[], items: K8sResourceKind[]) =>
+    ownerRefs.filter(({ uid }) => items.filter(({ metadata }) => metadata.uid === uid).length > 0);
+
+  return (resources: { [kind: string]: { data: K8sResourceKind[] } }) =>
+    _.flatMap(resources, (resource) => _.map(resource.data, (item) => item)).filter(
+      ({ kind, metadata }, i, allResources) =>
+        providedAPIs.filter((item) => item.kind === kind).length > 0 ||
+        owners(metadata.ownerReferences || [], allResources).length > 0,
+    );
+};
+const checkForNs = (obj: K8sResourceKind) => {
+  // Revisit later
+  const olmTargetNamespaces = obj?.metadata?.annotations?.['olm.targetNamespaces'] ?? '';
+  const managedNamespaces = olmTargetNamespaces?.split(',') || [];
+  return managedNamespaces.length === 1 && managedNamespaces[0] === '';
+};
+
+const rowFilters = (firehoseResource) =>
+  firehoseResource.length > 1
+    ? [
+        {
+          filterGroupName: 'Resource Kind',
```

Comment:
This might not be new, but we'd need to pull this into the render function to translate the group label.